### ISSUE TITLE
Adapt optimized models to work with SciUnit

### ISSUE THREAD

**Issue by lbologna** [cnr-ibf-pa/hbp-bsp-issues#35]

**Comment by alex4200** [comment]
@apdavison : Can you please give some feedback if this is already done, or by when you think you can finish this task?

**Comment by alex4200** [comment]
@apdavison : Can you please give some feedback if this is already done, or by when you think you can finish this task?

**Comment by apdavison** [comment]
I believe this is done. @appukuttan-shailesh can you give the details?

**Comment by appukuttan-shailesh** [comment]
I believe we are referring here to running validation tests on optimized hippocampal models. This has been completed a while ago (pending feedback for changes/enhancements), and can be [found here](https://github.com/apdavison/hippocampus_CA1_pyramidal). There are two ways of running the tests:

1. [Specifying a directory containing multiple optimized models as .zip files](https://github.com/apdavison/hippocampus_CA1_pyramidal/blob/master/run_tests_fitted_allzips.py)

2. [Individual model zip name specified via shell call](https://github.com/apdavison/hippocampus_CA1_pyramidal/blob/master/run_tests_fitted_shellcall.py)

Do let us know if this works and/or if anything needs to be changed.

**Comment by apdavison** [comment]
Leaving the ticket open until this is documented (Jupyter notebook?)

**Comment by alex4200** [comment]
@apdavison : Yes please leave this ticket open and let me know when and where this is documented. Thanks

**Comment by apdavison** [comment]
We will make this available as an Online Use Case

**Comment by alex4200** [comment]
Expected until about 21 Feb 2018

**Comment by appukuttan-shailesh** [comment]
The validation use case added last week is a specific variant (for Basal Ganglia models) of that intended here. The generic version proposed here would be ready by the end of next week.

**Comment by appukuttan-shailesh** [comment]
@lbologna @clupascu 
The optimized models stored at https://github.com/lbologna/bsp_data_repository earlier used to contain a single `cell.hoc` inside the checkpoints directory. The validation test code for these models used to load this file. In the optimized models now available on the repository, the models seem to have multiple `.hoc` files in the checkpoints directory. E.g.
For `CA1_pyr_cACpyr_mpg141017_a1-2_idC_20170918151638`, we have:
```
cell_seed3_0.hoc
cell_seed4_0.hoc
cell_seed5_0.hoc
cell_seed6_0.hoc
```
How do these files differ? What do the seeds refer to?
For the tests in HippoUnit (with Sara) we currently pick any one of these files ([see here](https://github.com/KaliLab/hippounit/blob/master/hippounit/utils.py#L737)).

Is it ok to validate for any ONE of these `.hoc` files? Or is something else expected?

**Comment by appukuttan-shailesh** [comment]
@alex4200: This use case is now complete. I have made a PR in the 'UseCases' repo... so you should be able to see it in the Validation UseCases once it is merged (named 'Generic - BluePyOpt Optimized Model Validation').

**Comment by appukuttan-shailesh** [comment]
@alex4200 : This is available for testing.

**Comment by alex4200** [comment]
Yes done
